drape_frontend/color_constants.cpp

```diff
@@ -1,15 +1,139 @@
 #include "drape_frontend/color_constants.hpp"
 #include "drape_frontend/apply_feature_functors.hpp"
 
+#include "platform/platform.hpp"
+
 #include "indexer/drawing_rules.hpp"
+#include "indexer/map_style_reader.hpp"
+
+#include "coding/reader.hpp"
 
 #include "base/assert.hpp"
+#include "base/string_utils.hpp"
+
+#include "3party/jansson/myjansson.hpp"
+
+#include <fstream>
+#include <map>
+
+namespace
+{
+class TransitColorsHolder
+{
+public:
+  dp::Color GetColor(std::string const & name) const
+  {
+    auto const style = GetStyleReader().GetCurrentStyle();
+    auto const isDarkStyle = style == MapStyle::MapStyleDark || style == MapStyle::MapStyleVehicleDark;
+    auto const & colors = isDarkStyle ? m_nightColors : m_clearColors;
+    auto const it = colors.find(name);
+    if (it == colors.cend())
+    {
+      LOG(LWARNING, ("Requested transit color '" + name + "' is not found"));
+      return dp::Color();
+    }
+    return it->second;
+  }
+
+  void Load()
+  {
+    std::string data;
+    try
+    {
+      ReaderPtr<Reader>(GetPlatform().GetReader("transit_colors.txt")).ReadAsString(data);
```

Comment:
Move the filename to a separate constant?

Reply:
Also, the kTransitColorPrefix constant may be reused.

Reply:
fixed